src/main/java/org/mitre/synthea/engine/State.java

```diff
@@ -1586,6 +1605,26 @@ public boolean process(Person person, long time) {
     @Override
     protected void initialize(Module module, String name, JsonObject definition) {
       super.initialize(module, name, definition);
+      validate(module, name);
+    }
+
+    protected void validate(Module module, String name) {
+      if (exact != null || range != null) {
+        // units are required
+        if (unit == null || unit.isEmpty()) {
+          throw new RuntimeException(
+              "Observations with numeric quantities must contain units matching pattern [^\\s]+: "
+          + "Module \"" + module.name + "\": State \"" + name + "\": Unit missing or empty");
+        }
+
+        String regex = "[^\\s]+";
```

Comment:
@dehall Where did the regex come from? I'm having a hard time understanding exactly what it is trying to do. Is it essentially ensuring that the string is not blank?

Reply:
That's from this commit on `cabg_cardio_gmf`: https://github.com/synthetichealth/synthea/commit/22c116bdada0024b087db61025a194a24b7a8d02

Looking at it now though it is a roundabout way of doing things. `\s` means any whitespace character, the double backslash is just for java string escaping, `^` means negate, the brackets `[]` mean "anything from this group", but I'm not sure it's meaningful when the one thing inside it is already a group, and `+` means one or more. So this will match anything that is one or more non-whitespace characters. 
It's a little roundabout because it should be equivalent to `"\\S+"` where the `\S` class means non-whitespace. 
( https://docs.oracle.com/javase/tutorial/essential/regex/pre_char_classes.html )
If we were using Java 11 we could use `String.isBlank` instead here, but that's not in 8. I can change this to use Apache `StringUtils.isBlank` if that's more clear. We already use StringUtils in a few other places so it's not a whole new library just for this